hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/KafkaHeaderNameProperties.java

```diff
@@ -14,12 +16,15 @@ public class KafkaHeaderNameProperties implements KafkaHeaderNameParameters {
 
     private String timestamp = "ts";
 
+    private Set<String> internalHeaders = Set.of(messageId, timestamp, schemaId, schemaVersion);
```

Comment:
`Set#of` creates an immutable collection so updating in `updateInternalHeaders` will throw an `UnsupportedOperationException`.  

We should switch to mutable set here or remove `updateInternalHaders` at all and implement `isNotInternal` as:

```java
public boolean isNotInternal(String name) {
        return Set.of(this.messageId, this.timestamp, this.schemaId, this.schemaVersion).contains(name);
    }
```

Reply:
also: the fact that it didn't blow up in integration tests mean that tests for this part are missing

Reply:
I agree that writing tests just for this class is not necessary. What I meant is that we have a piece of configuration that is never used is integration tests, so we never check if everything is wired correctly by spring and that properties are applied correctly. If we don't test all properties in e2e tests that we can only find out about bugs after deployment